src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+
+    const int base = 10;
+    long firstOctet = std::strtol(firstOctetStr.c_str(), NULL, base);
+    long secondOctet = std::strtol(secondOctetStr.c_str(), NULL, base);
+    long thirdOctet = std::strtol(thirdOctetStr.c_str(), NULL, base);
+    long fourthOctet = std::strtol(fourthOctetStr.c_str(), NULL, base);
+
+    const short maxIpValue = 255;
+    const short minIpValue = 0;
+    if (firstOctet > maxIpValue || secondOctet > maxIpValue || thirdOctet > maxIpValue || fourthOctet > maxIpValue)
+        return false;
+    if (firstOctet < minIpValue || secondOctet < minIpValue || thirdOctet < minIpValue || fourthOctet < minIpValue)
+        return false;
+
+    return true;
+}
+
+/**
+ * @brief Checks if the ip address is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid port. If that is true, the function returns.
+ * If that is not true, the function checks if it is a valid ip address and reads it.
+ *
+ */
+void ConfigFileParser::readIpAddress(const std::string& value)
+{
+    size_t colonIndex = value.find(':');
+    std::string ip;
+
+    if (colonIndex == std::string::npos) {
+        size_t semicolonIndex = value.find(';');
+        std::string num = value.substr(0, semicolonIndex);
+        if (isPortValid(num))
+            return;
+        if (value.find('.') == std::string::npos)
+            throw std::runtime_error("Invalid port");
+        ip = num;
+    } else
+        ip = value.substr(0, colonIndex);
+
+    if (!isIpAddressValid(ip))
+        throw std::runtime_error("Invalid ip address");
+
+    m_configFile.servers[m_configFile.serverIndex].listen.insert(std::make_pair(ip, 0));
+}
+
+/**
+ * @brief Checks if the value of the listen directive is valid and reads it if that is the case
+ *
+ * The function makes sure that the port is valid in the following ways:
+ * 1. The port must not contain a character other than '0'-'9'
+ * 2. The value of the port must be between 1-65535
+ *
+ * @param port The port to be checked
+ * @return true If the port is valid
+ * @return false If the port is invalid
+ */
+bool ConfigFileParser::isPortValid(const std::string& port) const
+{
+    if (port.find_first_not_of("0123456789") != std::string::npos)
+        return false;
+
+    const int base = 10;
+    const int maxPort = 65535;
+    const int minPort = 1;
+
+    int long portNum = std::strtol(port.c_str(), NULL, base);
+    return !(portNum <= minPort || portNum > maxPort);
+}
+
+/**
+ * @brief Checks if the port is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid ip address. If that is true, the function returns.
+ * If that is not true, the function and checks if it is a valid port and reads it.
+ *
+ */
+void ConfigFileParser::readPort(const std::string& value)
```

Comment:
Agree. 

Created one readSocket functions which reads ip address and port instead of two separate functions. 
Furthermore the port will now be stored as a string